### PR TITLE
Fix a regression causing "started from" without title to appear on new cards

### PR DESCRIPTION
**Pull request by attekei**
This regression was caused by #4957. This appears when you create a new card and modify the original query or run a drill-through action. 

I added also some comments to better explain what is the purpose of `setCardAndRun`.

### CODE REVIEW

**Comment by tlrobinson** [comment]
I don't think it was due to this PR, but it looks like at some point we started immediately updating the  URL to include the serialized card for saved questions, even before the question is modified, which we shouldn't do. i.e. `/question/1` should remain `/question/1` until the user changes something about the card. We shouldn't immediately change to `/question/1#eyJuYW1lIjoiT3JkZXJzLCBD...`.

This also seems like an area that would benefit a lot from testing.

**Comment by attekei** [comment]
True! I will fix `isCardDirty` and add comprehensive tests for it.

**Comment by attekei** [comment]
`isCardDirty` is now fixed and has tests.

File: frontend/src/metabase/lib/card.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ export function isCardDirty(card, originalCard) {
         }
     } else {
         const origCardSerialized = originalCard ? serializeCardForUrl(originalCard) : null;
-        const newCardSerialized = card ? serializeCardForUrl(card) : null;
+        const newCardSerialized = card ? serializeCardForUrl(_.omit(card, 'original_card_id')) : null;
```
Comment (tlrobinson) [inline frontend/src/metabase/lib/card.js]: Can we just have `serializeCardForUrl` not include `original_card_id`, or does it need to be included in some cases?

Comment (attekei) [inline frontend/src/metabase/lib/card.js]: It's required for fetching the original card info and showing the lineage if you later return to a card through browser history.

**Review by tlrobinson — APPROVED** [review]

**Pull request merged**